### ISSUE TITLE
Install error: relocation R_X86_64_PC32 against `lobjectType'

### ISSUE THREAD

**Issue by enquea** [psycopg/psycopg2#305]
Trying to install on Redhat 5, get this error:

/usr/bin/ld: build/temp.linux-x86_64-2.7/psycopg/psycopgmodule.o: relocation R_X86_64_PC32 against `lobjectType' can not be used when making a shared object; recompile with -fPIC

/usr/bin/ld: final link failed: Bad value

collect2: ld returned 1 exit status

error: command 'gcc' failed with exit status 1

*forgot to mention, this is from pip install on RHEL5

**Comment by dvarrazzo** [comment]
Check [this answer](http://stackoverflow.com/questions/16841674/python-c-extension-link-with-a-custom-shared-library) to find a way to pass -fPIC to the linker.

What libpq version are you compiling psycopg with? Could it be that it is *very* old?

**Comment by enquea** [comment]
Thanks for getting back to me.

Libpq is postgresql-lib on redhat right?

The version I have is 8.1.23

I'm able to install psycopg2.5, so I can just go with that for now.

**Comment by dvarrazzo** [comment]
I see: libpq 8.1 is way out of date and unsupported. While I still check against PG 8.1 server I only compile and test the module with more up-to-date client libraries.

**Comment by kamushin** [comment]
@wenbochang 
I got the same error information when I try to install psycopg2-2.6.1. It seems you succeeded in installing  psycopg2.5. Is here some trick?

**Comment by wenbochang** [comment]
@kamushin I'm installing inside pip requirements.txt, so I pointed the version specifically to 2.5

**Comment by shimizukawa** [comment]
I got the same error with using [manylinux1 docker container](https://www.python.org/dev/peps/pep-0513/#docker-image) that is based on CentOS 5.11.
psycopg2-2.5.x is fine to build. 2.6.x is not.

**Comment by dan-blanchard** [comment]
@shimizukawa It's because that docker container uses a very old version of libpq by default. We've worked around it by forking it and adding [these steps](https://github.com/Parsely/manylinux/blob/c2d4fe68fd8a18e74d2401204dd57b7fe9ee2add/docker/build_scripts/build.sh#L54-L58) to the build script, and then making it install `postgresql95-libs` and `postgresql95-devel`. We also had to set `PG_HOME` and `PATH` in our build script that builds all our wheels.
